From 3957a82301206e176bcd544a60d4fac3650f582c Mon Sep 17 00:00:00 2001
From: Yan Wang <yan.wang@linux.intel.com>
Date: Mon, 19 Mar 2018 21:02:47 +0800
Subject: [PATCH 1326/2313] ASoC: APL/CNL: Add stream related definition for
 DMA trace on APL/CNL.

On APL/CNL, DMA trace is transferred by one capture stream based
on host DMA. So need define stream related variable for it.

Signed-off-by: Yan Wang <yan.wang@linux.intel.com>
---
 include/uapi/sound/sof-ipc.h | 1 +
 sound/soc/sof/sof-priv.h     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/uapi/sound/sof-ipc.h b/include/uapi/sound/sof-ipc.h
index ec34ad0..1e24f27 100644
--- a/include/uapi/sound/sof-ipc.h
+++ b/include/uapi/sound/sof-ipc.h
@@ -809,6 +809,7 @@ struct sof_ipc_window {
 struct sof_ipc_dma_trace_params {
 	struct sof_ipc_hdr hdr;
 	struct sof_ipc_host_buffer buffer;
+	uint32_t stream_tag;
 }  __attribute__((packed));
 
 /* DMA for Trace params info - SOF_IPC_DEBUG_DMA_PARAMS */
diff --git a/sound/soc/sof/sof-priv.h b/sound/soc/sof/sof-priv.h
index 5a9f19b..ea68f4e 100644
--- a/sound/soc/sof/sof-priv.h
+++ b/sound/soc/sof/sof-priv.h
@@ -371,6 +371,7 @@ struct snd_sof_dev {
 	/* DMA for Trace */
 	struct snd_dma_buffer dmatb;
 	struct snd_dma_buffer dmatp;
+	struct snd_sof_hda_stream *dtrace_stream;
 	int dma_trace_pages;
 	wait_queue_head_t trace_sleep;
 	u32 host_offset;
-- 
2.7.4

